logstash-core/src/test/java/org/logstash/config/ir/CompiledPipelineTest.java

```diff
@@ -159,12 +164,57 @@ public void buildsForkedPipeline() throws Exception {
                 filters,
                 Collections.singletonMap("mockoutput", mockOutputSupplier())
             )
-        ).buildExecution().compute(RubyUtil.RUBY.newArray(testEvent), false, false);
+        ).buildExecution(false).compute(RubyUtil.RUBY.newArray(testEvent), false, false);
         final Collection<JrubyEventExtLibrary.RubyEvent> outputEvents = EVENT_SINKS.get(runId);
         MatcherAssert.assertThat(outputEvents.size(), CoreMatchers.is(1));
         MatcherAssert.assertThat(outputEvents.contains(testEvent), CoreMatchers.is(true));
     }
 
+    @Test
+    public void preservesInputEventInstantiationOrder() throws Exception {
+        testEventOrder(true, new int[]{0, 2, 1});
+    }
+
+    @Test
+    public void ignoresInputEventInstantiationOrder() throws Exception {
+        testEventOrder(false, null);
```

Comment:
this test assert that event instantiation is _ignored_, but then goes on to test that an out-of-order batch has its ordering preserved.

I don't feel that this matches well to the guarantees that we are attempting to make.

Reply:
This tests that the overhead of event ordering is _not_ present when not requested. Do you think that this doesn't need to be tested or that it should be tested differently?

Reply:
This tests verifies the current stable observed behaviour of event re-ordering caused by this specific pipeline's structure, which means it is susceptible to failing when unrelated changes happen. Since we make _no_ ordering guarantees in this case, validating a specific out-of-order ordering with a test named to indicate that it _ignores_ ordering feels like a mismatch.

I would be fine with the test being named something like `doesNotIncurEventOrderingOverhead` along with a comment indicating that the ordering we are checking is the observed (and stable) behaviour of events coming into the sync from the pipeline as-defined.